trabajo-final/spotify_assistant/src/Reducer.js

```diff
@@ -0,0 +1,49 @@
+import { REHYDRATE } from 'redux-persist/constants'
+
+const initialState = {
+  user: null,
+  playlistEditting: [],
+  spotify: null,
+  searchedTracks: [],
+  localPlaylist: [],
+};
+
+export default function reducer(state = initialState,action) {
+  switch (action.type) {
+    case 'REFRESH_EDITTING':
+      return Object.assign({},state,{playlistEditting: []});
+    case 'SPOTIFY':
+      return Object.assign({},state,{spotify: action.spotify});
+    case 'ADD_USER_DATA':
+      return Object.assign({},state,{user: action.user});
+    case 'NEW_SEARCH':
+      return Object.assign({},state,{searchedTracks: action.tracks});
+    case 'ADD_TRACK_TO_PLAYLIST':
+      return Object.assign({},state,{playlistEditting: [...state.playlistEditting, action.track]});
+    case 'ADD_PLAYLIST_TO_LOCAL':
+      return Object.assign({},state,{localPlaylist:[...state.localPlaylist,action.playlist]});
+    case 'DELETE_LOCAL_PLAYLIST':
+      let index = 0;
+      for(index in state.localPlaylist){
```

Comment:
https://www.w3schools.com/jsref/jsref_find.asp

Take a look at this. It will improve this iteration